features/guidance/turn-angles.feature

```diff
@@ -0,0 +1,1104 @@
+            """
+                              f - - - - - - - - - - - - - - - - - - - - e
+                             '
+                            '
+                           '
+                          '
+                         '
+            a - - - - - b
+                         '
+                          '
+                           '
+                            '
+                             '
+                              c - - - - - - - - - - - - - - - - - - - - d
+            """
+
+        And the ways
+            | nodes | highway | name | lanes | oneway |
+            | ab    | primary | road | 4     | no     |
+            | bcd   | primary | road | 2     | yes    |
+            | efb   | primary | road | 2     | yes    |
+
+        When I route I should get
+            | waypoints | route     | turns         |
+            | a,d       | road,road | depart,arrive |
+            | e,a       | road,road | depart,arrive |
+
+    @todo
+    # currently the intersections don't match up do to the `merging` process.
+    # The intermediate intersection is technically no-turn at all, since the road continues.
+    Scenario: Splitting Road with many lanes
+        Given the node map
+            """
+                              f - - - - - - - - - - - - - - - - - - - - e
+                             '
+                            '
+                           '
+                          '
+                         '
+            a - - - - - b
+                         '
+                          '
+                           '
+                            '
+                             '
+                              c - - - - - - - - - - - - - - - - - - - - d
+            """
+
+        And the ways
+            | nodes | highway | name | lanes | oneway |
+            | ab    | primary | road | 4     | no     |
+            | bcd   | primary | road | 2     | yes    |
+            | efb   | primary | road | 2     | yes    |
+
+        When I route I should get
+            | waypoints | route     | turns         | intersections                                |
+            | a,d       | road,road | depart,arrive | true:90,false:45 true:135 false:270;true:270 |
+            | e,a       | road,road | depart,arrive | true:270,false:45 true:135 true:270;true:90  |
+
+
+    #http://www.openstreetmap.org/#map=19/52.54759/13.43929
+    Scenario: Curved Turn At Cross
+        Given the node map
+            """
+                                                h
+                                                |
+                                                |
+                                                |
+                                                |
+                                                |      . b - - - - - - - - - - - - - - - - - - a
+                                                |    .
+                                                |  .
+                                                | c
```

Comment:
Clashes can occur, of course. But the worst that can happen is issuing `turn right` for both, which is (in combination with the intersection classification) a valid indication.

Still, you would have roads more or less overlapping, the road is only 3 meters wide here after all.